internal/jobobject/limits.go

```diff
@@ -0,0 +1,266 @@
+package jobobject
+
+import (
+	"context"
+	"fmt"
+	"unsafe"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	processorWeightMax float64 = 10000
+
+	// MemoryLimitMax is the max memory limit a job object can be assigned
+	MemoryLimitMax uint64 = 0xffffffffffffffff
+)
+
+func isFlagSet(flag uint32, controlFlags uint32) bool {
+	if (flag & controlFlags) == flag {
+		return true
+	}
+	return false
+}
+
+// SetMemoryLimit sets the memory limit the job object represented by `jobHandle`
+// based on the given `memoryLimitInBytes`
+func SetMemoryLimit(jobHandle windows.Handle, memoryLimitInBytes uint64) error {
```

Comment:
Exporting versions of these functions that take a `windows.Handle` directly kind of breaks the encapsulation that this package aims to provide. Given this package defines a type that is intended to be a "job object", I would ask why we can't use that type, rather than operating on handles directly?

Reply:
(I suspect it is because there is no `jobobject.Open` func today, but then I would ask if we can't add one instead of exposing internal details of the package)

Reply:
Didn't see this comment before writing mine haha Agreed on this

Reply:
I added this, good suggestion, thank you :)